tests/unit/dataset/dataset_definition_test.py

```diff
@@ -18,76 +18,289 @@
 # OUT OF OR IN CONNECTION WITH THE SOFTWARE OR THE USE OR OTHER DEALINGS IN THE
 # SOFTWARE.
 """Test dataset definition."""
-from dataclasses import asdict
 from dataclasses import dataclass
-from dataclasses import field
 
+import pytest
 import yaml
 
 from pymovements import DatasetDefinition
 from pymovements import DatasetLibrary
 from pymovements import Experiment
 
 
-def test_dataset_definition_to_yaml_equal_dicts_no_exp(tmp_path):
-    tmp_file = tmp_path / 'tmp.yaml'
+@pytest.mark.parametrize(
+    ('definition', 'expected_dict'),
+    [
+        pytest.param(
+            DatasetDefinition(
+                name='Example',
+                has_files={
+                    'gaze': False,
+                    'precomputed_events': False,
+                    'precomputed_reading_measures': False,
+                },
+            ),
+            {
+                'name': 'Example',
+                'has_files': {
+                    'gaze': False,
+                    'precomputed_events': False,
+                    'precomputed_reading_measures': False,
+                },
+                'acceleration_columns': None,
+                'column_map': {},
+                'custom_read_kwargs': {},
+                'distance_column': None,
```

Comment:
The output of to_dict() currently includes a lot of attributes that are just None or empty dictionaries.
I created https://github.com/aeye-lab/pymovements/issues/1091 to fix this.